tests/dialects/test_memref.py

```diff
@@ -293,3 +297,25 @@ def test_memref_cast():
 
     assert cast.source is memref_ssa_value
     assert cast.dest.typ is res_type
+
+
+def test_memref_dma_wait():
+    tag_type = MemRefType.from_element_type_and_shape(i32, [4])
```

Comment:
Does this work?
```suggestion
    tag_type = MemRefType(i32, [4])
```

Reply:
Doesn't work. Removed the commit that changed it.